### PR TITLE
DDP-6625 make groups in autocomplete picklist unselectable

### PR DESCRIPTION
**Pull request by aglushko-broad**

### CODE REVIEW

**Comment by MocanaAtBroad** [comment]
Since we are matching group names as we type, we probably should underline the matching group name substring?
Screen Shot 2021-07-27 at 9 07 46 AM

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts [hunk 1]
```diff
@@ -57,8 +53,8 @@ export class AutocompleteActivityPicklistQuestion extends BaseActivityPicklistQu
     public ngOnInit(): void {
         const answer = this.block.answer && this.block.answer[0];
         if (answer) {
-            this.inputFormControl
-                .setValue(answer.detail || this.block.picklistSuggestions.find(option => option.value === answer.stableId));
+            this.inputFormControl.setValue(answer.detail || this.block.picklistGroups.flatMap(group => group.options)
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts:56]: Wouldn't this assignment end up setting a a value of type ActivityPicklistOption and wouldn't that be a problem?

**Comment by aglushko-broad** [comment]
> Since we are matching group names as we type, we probably should underline the matching group name substring?
> Screen Shot 2021-07-27 at 9 07 46 AM

done

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts (hunk 1 above)
Comment (MocanaAtBroad) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts:56]: I see. Value can be a string or an iActivityPicklistOption.
Would it be simpler to convert values to strings and look up the option just when you have to? Just seeing lot of checks of string types and "string | ActivityPicklistOption"  type definitions?
(This a question/something to consider, not a request for a change).

File: ddp-workspace/projects/toolkit/tsconfig.lib.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "types": [],
     "lib": [
       "dom",
-      "es2018"
+      "es2019"
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/toolkit/tsconfig.lib.json:17]: Presumably for flatMap? Make sure you can `ng build` the pancan project.

**Review by MocanaAtBroad — COMMENTED** [review]
Code is clean and works nice. Just a couple of comments that might or might not require changes.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts (hunk 1 above)
Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts:56]: we need to use the object for auocomplete model. Label is for displaying and stableId is a value. When user picks something from the list we send the whole object that contains the right id. It's faster than search for the id based on the label, so I'd prefer to not change it.
I think it's fine having unit types when we work with such controls like autocmplete with custom value

File: ddp-workspace/projects/toolkit/tsconfig.lib.json (hunk 2 above)
Comment (aglushko-broad) [reply ddp-workspace/projects/toolkit/tsconfig.lib.json:17]: Correct! Yes, it works w/o errors :)

**Review by alextanlan — APPROVED** [review]

**Pull request merged**